### PR TITLE
Implement Next Leader and the Rest of Leader for DA consensus

### PR DESCRIPTION
**Pull request by bfish713**
Adding logic for The leader and next leader in the DA/Sequencing consensus
- The `DAConsensusLeader` is the task which builds and broadcast the `CommitmentProposal`.  For now this tasks depends on the DAProposal succeeding and can only begin after that.  
- DANextLeader is the next leader task which collects votes on the previous leaders proposal and builds the QC

The vote collection logic for `DAProposals` and `CommitmentProposals` is quite similar and I intend to refactor to use the new election trait after this.  It should reduce the duplicated code to some degree

Also I stubbed out the network function for sending the DA proposal.  Currently, the networking traits all assume 1 type of proposal but we need to send DA and Commitment proposals.  I stubbed it out because this exactly what @DieracDelta is working on with the networking trait refactor.  

Closes: https://github.com/EspressoSystems/HotShot/issues/715

### CODE REVIEW

File: consensus/src/traits.rs [hunk 1]
```diff
@@ -152,6 +152,12 @@ pub trait ConsensusApi<
         .await;
     }
 
+    /// Send a broadcast to the DA comitee, stub for now
+    async fn send_da_broadcast<DAPROPOSAL: ProposalType<NodeType = TYPES>>(
```
Comment (DieracDelta) [inline consensus/src/traits.rs:156]: I think this might need a view number to determine the DA committee for this view.

File: src/lib.rs [hunk 2]
```diff
@@ -995,6 +1035,23 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>>
             .await
     }
 
+    async fn send_da_broadcast<DAPROPOSAL: ProposalType<NodeType = TYPES>>(
+        &self,
+        _message: ConsensusMessage<TYPES, I::Leaf, DAPROPOSAL>,
+    ) -> std::result::Result<(), NetworkError> {
+        // TODO: Should look like this code but it won't work due to only 1 Proposal type being
+        // associated with self.inner.networking.
```
Comment (DieracDelta) [inline src/lib.rs:1043]: Makes sense! Like you said, the DA networking changes should help with this.

**Review by DieracDelta — APPROVED** [review]

File: consensus/src/traits.rs (hunk 1 above)
Comment (bfish713) [reply consensus/src/traits.rs:156]: I will add view number to the function signature as a placeholder for now.

File: consensus/src/da.rs [hunk 3]
```diff
@@ -194,33 +267,202 @@ where
...
+            state
+        } else {
+            warn!("Don't have last state on parent leaf");
+            return None;
+        };
+        if let Ok(new_state) = starting_state.append(&self.block, &self.cur_view) {
+            let leaf = DALeaf {
+                view_number: self.cur_view,
+                // TODO: what is this height for?
+                height: 0,
```
[the remark this answers is not in the document]
Comment (bfish713) [reply consensus/src/da.rs:347]: I also am not sure

**Pull request merged**